### PR TITLE
Transfer logout button from tabviews to owner and sitter profile screens

### LINKED ISSUE #33: Move `logout` button from tabviews to owner and sitter profile screens
## TO DISCUSS:
think about UI, where will be better to place this button

## Description
Remove `Logout` button from the navigation bar to ` Profile` tab

## UI Mocks
Screenshot 2023-10-01 at 12 43 17

### PR DESCRIPTION
**Pull request by iBYarchitect**
## Summary
This PR is a part of the work to resolve #33 

## Changes
- Created ContentView for switching between sitter, owner flow and default start screen.
- SitterProfileView
- OwnerProfileView
- SitterProfileViewModel
- OwnerProfileViewModel
- OwnerMainTabView
- SitterMainTabView
- OwnerWalkingsView
- SitterWalkingsView

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift [hunk 1]
```diff
@@ -5,105 +5,132 @@ struct SitterProfileView: View {
...
+        }
     }
 
     // MARK: - Private interface
 
     /// View model responsible to manage data from model layer.
     @StateObject private var viewModel = SitterProfileViewModel()
+
+    @EnvironmentObject private var userRoleViewModel: UserRoleViewModel
+    @Environment(\.dismiss) private var dismiss
```
Comment (kirpichonok) [inline Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:121]: How do we intend to use this?

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 2]
```diff
@@ -9,39 +9,84 @@ struct OwnerProfileView: View {
...
+                            address: viewModel.address,
+                            avatarUrl: viewModel.avatarURL
+                        )
+
+                        Button(isEditingMode ? saveButtonLabelText : editButtonLabelText) {
+                            if isEditingMode {
+                                viewModel.save()
+                            }
+                            isEditingMode.toggle()
+                        }
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:49]: What changes if to do in this way? 
```suggestion
                        Button(editButtonLabelText) {
                            isEditingMode.toggle()
                        }
```

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 3]
```diff
@@ -9,39 +9,84 @@ struct OwnerProfileView: View {
...
+                            phone: $viewModel.phone,
+                            address: $viewModel.address
+                        )
+
+                        Button(isEditingMode ? saveButtonLabelText : editButtonLabelText) {
+                            if isEditingMode {
+                                viewModel.save()
+                            }
+                            isEditingMode.toggle()
+                        }
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:28]: What changes if to do in this way?
```suggestion
                        Button( saveButtonLabelText) {
                            viewModel.save()
                            isEditingMode.toggle()
                        }
```

**Review by kirpichonok — CHANGES_REQUESTED** [review]
It's a huge work! It would be odd if I don't have questions ) Please, consider them.

The issue talk about:
> Move logout button from tabviews to owner and sitter
But I see these buttons on their previous places and I can't understand a purpose of this PR from the description.

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 4]
```diff
@@ -9,39 +9,84 @@ struct OwnerProfileView: View {
...
+                    .cornerRadius(AppStyle.UIElementConstant.cornerRadius)
                 }
-                isEditingMode.toggle()
+
+                Spacer()
             }
-            .buttonStyle(CapsuleWithWhiteText())
-            .padding()
             .padding(.horizontal)
-            .disabled(isEditingMode && viewModel.name.isEmpty)
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:42]: Yes, you are right.
Returned with 33b5f54

**Review by NadzeyaShpakouskaya — CHANGES_REQUESTED** [review]
Need to think about how to store the headers after moving the navigation stack inside the tab views.
It looks like we don't need more of the next files:

- SitterMainTabView+Tab
- OwnerMainTabView+Tab

Also need to think about the repeating strings like text in alerts and button text for both flows such as:
```swift
    private let alertLogOutTitle = "Do you really want to log out?"
    private let navigationTitle = "My profile"

    private let cancelButtonLabelText = "Cancel"
    private let continueButtonLabelText = "Continue"
    private let saveButtonLabelText = "Save"
    private let editButtonLabelText = "Edit"
    private let logoutButtonLabelText = "Logout"
```
Do we need to create the enum for some strings that repeat over the screens

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 1 above)
Comment (kirpichonok) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift:121]: I can still see this line in the same place.

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 5]
```diff
@@ -20,10 +20,8 @@ struct OwnerProfileView: View {
...
                         )
 
-                        Button(isEditingMode ? saveButtonLabelText : editButtonLabelText) {
-                            if isEditingMode {
-                                viewModel.save()
-                            }
+                        Button(saveButtonLabelText) {
+                            viewModel.save()
                             isEditingMode.toggle()
                         }
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:26]: I think this line must be here.
```suggestion
                        }
                        .disabled(isEditingMode && viewModel.name.isEmpty)
```

**Review by kirpichonok — CHANGES_REQUESTED** [review]
It seems like you didn't strive 😄

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift (hunk 5 above)
Comment (iBYarchitect) [reply Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:26]: Fixed with Fixed with 53bbc79

File: Woof/Woof/View/SitterFlow/SitterWalkingsView/SitterWalkingsView.swift [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import SwiftUI
+
+/// A view displaying information about the pet walkings.
+struct SitterWalkingsView: View {
+    var body: some View {
+        NavigationView {
+            Text("Walkings stub")
+                .navigationTitle("Walkings")
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/View/SitterFlow/SitterWalkingsView/SitterWalkingsView.swift:8]: Fixed with 53bbc79

File: Woof/Woof/View/OwnerFlow/OwnerWallkingsView/OwnerWalkingsView.swift [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import SwiftUI
+
+/// A view displaying information about the pet walkings.
+struct OwnerWalkingsView: View {
+    var body: some View {
+        NavigationView {
+            Text("Walkings stub")
+                .navigationTitle("Walkings")
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/View/OwnerFlow/OwnerWallkingsView/OwnerWalkingsView.swift]: Fixed with 53bbc79

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 8]
```diff
@@ -9,39 +9,80 @@ struct OwnerProfileView: View {
...
+                }
+                Button(
+                    cancelButtonLabelText,
+                    role: .cancel
+                ) { viewModel.isAlertShown.toggle() }
+            }
         }
     }
+
+    @EnvironmentObject private var userRoleViewModel: UserRoleViewModel
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:78]: Fixed with 53bbc79

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 9]
```diff
@@ -9,39 +9,80 @@ struct OwnerProfileView: View {
...
+                    .cornerRadius(AppStyle.UIElementConstant.cornerRadius)
                 }
-                isEditingMode.toggle()
+
+                Spacer()
             }
-            .buttonStyle(CapsuleWithWhiteText())
-            .padding()
             .padding(.horizontal)
             .disabled(isEditingMode && viewModel.name.isEmpty)
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift]: Fixed with 53bbc79

File: Woof/Woof/ContentView.swift [hunk 10]
```diff
@@ -0,0 +1,20 @@
+import SwiftUI
+
+/// A root view for the app .
+struct ContentView: View {
+    @StateObject var userRoleVM = UserRoleViewModel()
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/ContentView.swift]: Fixed with 53bbc79

File: Woof/Woof/ContentView.swift [hunk 11]
```diff
@@ -0,0 +1,20 @@
+import SwiftUI
+
+/// A root view for the app .
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/ContentView.swift]: Fixed with 53bbc79

File: Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift (hunk 1 above)
Comment (iBYarchitect) [reply Woof/Woof/View/SitterFlow/SitterProfileView/SitterProfileView.swift]: Which line do you still see?

**Comment by iBYarchitect** [comment]
Added enums with text descriptions for buttons and alerts in the app.

File: Woof/Woof/Helper/AppAlert/AppAlert.swift [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+    case logOut = "Do you really want to log out?"
+
+    /// Alert shown as a result of an error
+    case error = "Error"
+
+    /// Text description of alert
+    var description: String {
+        rawValue
+    }
+}
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Helper/AppAlert/AppAlert.swift:15]: Let's create the new PR for this enhancement, as we need to add unit tests for new enums and the current PR is already so big.

File: Woof/Woof/Helper/AppButton/AppButton.swift [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+/// Represents name of buttons which uses in application.
+enum AppButton: String, CustomStringConvertible {
+    case cancel = "Cancel"
+    case save = "Save"
+    case edit = "Edit"
+    case tryAgain = "Try Again"
+    case logout = "Logout"
+    case `continue` = "Continue"
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Helper/AppButton/AppButton.swift:10]: Missing doc comments, as mentioned above let's create separate PR for this as well

File: Woof/Woof/View/SitterFlow/ScheduleView/ScheduleView.swift [hunk 14]
```diff
@@ -4,11 +4,15 @@ import SwiftUI
 struct ScheduleView: View {
     var body: some View {
         NavigationView {
-            Text("My schedule")
-                .navigationTitle("My Schedule")
+            Text(navigationTitle)
+                .navigationTitle(navigationTitle)
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/View/SitterFlow/ScheduleView/ScheduleView.swift:8]: Do we use the same value for both cases?

**Review by NadzeyaShpakouskaya — APPROVED** [review]
Overall LGTM after resolving all conversation

File: Woof/Woof/View/SitterFlow/ScheduleView/ScheduleView.swift (hunk 14 above)
Comment (iBYarchitect) [reply Woof/Woof/View/SitterFlow/ScheduleView/ScheduleView.swift]: Fixed with 2afb5c0

File: Woof/Woof/Helper/AppButton/AppButton.swift (hunk 13 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: Fixed with 19ade1f

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 12 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Fixed with 19ade1f

**Pull request merged**